plugins/ipam/host-local/backend/disk/backend.go

```diff
@@ -93,28 +97,73 @@ func (s *Store) LastReservedIP(rangeID string) (net.IP, error) {
 	return net.ParseIP(string(data)), nil
 }
 
+// Release will mark a given ip as released, regardless of which container holds it.
 func (s *Store) Release(ip net.IP) error {
 	return os.Remove(GetEscapedPath(s.dataDir, ip.String()))
 }
 
+// ReleaseByID releases all IPs held by a container of the given id.
 // N.B. This function eats errors to be tolerant and
 // release as much as possible
-func (s *Store) ReleaseByID(id string) error {
-	err := filepath.Walk(s.dataDir, func(path string, info os.FileInfo, err error) error {
-		if err != nil || info.IsDir() {
+//
+func (s *Store) ReleaseByID(id backend.Key) error {
+	err := s.walkReservations(func(path string, _ net.IP, b []byte) error {
+		match, err := Matches(&id, b)
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "WARN: host-local: Failed to parse ip reservation file %s: %v", path, err)
 			return nil
 		}
-		data, err := ioutil.ReadFile(path)
+		if match {
+			err := os.Remove(path)
+			if err != nil {
+				fmt.Fprintf(os.Stderr, "WARN host-local: failed to remove ip reservation file %s: %v", path, err)
+			}
+		}
+		return nil
+	})
+	return err
+}
+
+// GetByID returns all IPs reserved by a given container ID
```

Comment:
"container ID" is a specific thing, and we are using ID+interface (with some fuzzing to cope with older stores).  Maybe "a given key" ?